lighty-modules/lighty-gnmi/lighty-gnmi-sb/README.md

```diff
@@ -55,3 +65,164 @@ This encoding enforcement does not apply to SetRequest/GetResponse targeting sca
+Creating device connection and All CRUD operation will be performed with writing data directly to MD-SAL data store
+without using RESTCONF. Full example can be found inside test [GnmiWithoutRestconfTest](../lighty-gnmi-test/src/test/java/io/lighty/modules/gnmi/test/gnmi/GnmiWithoutRestconfTest.java).
+
+1. Initialize lighty Controller lighty gNMI.
+```java
+        lightyController = new LightyControllerBuilder()
+                .from(ControllerConfigUtils.getConfiguration(Files.newInputStream(CONFIGURATION_PATH)))
+                .build();
+        lightyController.start().get();
+
+        gnmiSouthboundModule = new GnmiSouthboundModuleBuilder()
+                .withConfig(GnmiConfigUtils.getGnmiConfiguration(Files.newInputStream(CONFIGURATION_PATH)))
+                .withLightyServices(lightyController.getServices())
+                .withExecutorService(Executors.newCachedThreadPool())
+                .withEncryptionService(createEncryptionService())
+                .build();
+        gnmiSouthboundModule.start().get();
+```
+
+2. Connect gNMI device. More about configuring security or extension parameter on gNMI device can be found in
+[RCgNMI Documentation](https://github.com/PANTHEONtech/lighty/blob/master/lighty-applications/lighty-rcgnmi-app-aggregator/README.md#how-to-use-rcgnmi-example-app)
+and in [gnmi-topology.yang](https://github.com/PANTHEONtech/lighty/blob/master/lighty-models/lighty-gnmi-models/lighty-gnmi-topology-model/src/main/yang/gnmi-topology.yang)
+YANG model.
+```java
+        final Node testGnmiNode = createNode(GNMI_NODE_ID, DEVICE_ADDRESS, DEVICE_PORT, getInsecureSecurityChoice());
+        final WriteTransaction writeTransaction = bindingDataBroker.newWriteOnlyTransaction();
+        final InstanceIdentifier<Node> nodeInstanceIdentifier = IdentifierUtils.gnmiNodeIID(testGnmiNode.getNodeId());
+        writeTransaction.put(LogicalDatastoreType.CONFIGURATION, nodeInstanceIdentifier, testGnmiNode);
+        writeTransaction.commit().get();
+```
+
+3. Wait until gNMI device will be successfully connected to gNMI-sb
+```java
+        Awaitility.waitAtMost(WAIT_TIME_DURATION)
+                .pollInterval(POLL_INTERVAL_DURATION)
+                .untilAsserted(() -> {
+                    final Optional<Node> node = readOperData(bindingDataBroker, nodeInstanceIdentifier);
+                    assertTrue(node.isPresent());
+                    final Node foundNode = node.get();
+                    final GnmiNode gnmiNode = foundNode.augmentation(GnmiNode.class);
+                    assertNotNull(gnmiNode);
+                    final NodeState nodeState = gnmiNode.getNodeState();
+                    assertNotNull(nodeState);
+                    final NodeState.NodeStatus nodeStatus = nodeState.getNodeStatus();
+                    assertEquals(NodeState.NodeStatus.READY, nodeStatus);
+                });
+```
```

Comment:
change this to show the reader just how to read connection status and when a device is connected (`NodeState.NodeStatus.READY`)

Reply:
done